src/generic_datatype.cpp

```diff
@@ -135,20 +135,25 @@ int GenericDatatype::get_num_selectors(std::string cons) const
 {
   // Used to keep track of the number of selectors in the constructor
   int num_selectors = 0;
+  bool found = false;
   for (unsigned int i = 0; i < cons_decl_vector.size(); ++i)
   // Searches for a matching constructor
   {
     if (static_pointer_cast<GenericDatatypeConstructorDecl>(cons_decl_vector[i])
             ->get_name()
         == cons)
     {
+      found = true;
       // Calls the constructor's get_selector_count() function
       num_selectors = static_pointer_cast<GenericDatatypeConstructorDecl>(
                           cons_decl_vector[i])
                           ->get_selector_count();
```

Comment:
This is somewhat unrelated to the PR but can't we `break` once we found the constructor?